odbc/src/api/functions.rs

```diff
@@ -748,10 +748,7 @@ pub unsafe extern "C" fn SQLColAttributeW(
                 None => {
                     let mongo_handle = MongoHandleRef::from(statement_handle);
                     let _ = must_be_valid!((*mongo_handle).as_statement());
-                    add_diag_info!(
-                        mongo_handle,
-                        ODBCError::UnsupportedFieldDescriptor(format!("{field_identifier}"))
-                    );
+                    add_diag_info!(mongo_handle, ODBCError::DriverNotCapable);
```

Comment:
You flipped the errors.
If the descriptor is valid (e.i in the list), but we don't support it, we should return HYC00-`DriverNotCapable`
If it is not a know value, we should return HY091 - `Invalid descriptor` (code assigned to UnsupportedFieldDescriptor)